cmd/geth/main.go

```diff
@@ -425,8 +425,10 @@ func console(ctx *cli.Context) {
 	startNode(ctx, node)
 
 	// Attach to the newly started node, and either execute script or become interactive
-	client := utils.NewInProcRPCClient(node)
-
+	client, err := utils.NewRemoteRPCClientFromString("ipc:" + node.IpcEndpoint())
+	if err != nil {
+		utils.Fatalf("Failed to attach to the inproc geth: %v", err)
```

Comment:
If a user wants to run the console it requires the IPC interface to be active (can be disabled with `--ipcdisable`). This dependency is a bit strange. I think there is a valid use case for users who would like to run the console without any RPC interface enabled. E.g. platforms which don't have support for unix sockets/named pipes, or the user has no permissions to create one (SELinux).